src/ORM/DataObject.php

```diff
@@ -2241,7 +2241,15 @@ public function scaffoldSearchFields($_params = null)
 
             // Otherwise, use the database field's scaffolder
             } elseif ($object = $this->relObject($fieldName)) {
-                $field = $object->scaffoldSearchField();
+                if (is_object($object) && $object->hasMethod('scaffoldSearchField')) {
+                    $field = $object->scaffoldSearchField();
+                } else {
+                    throw new Exception(sprintf(
+                        "SearchField '%s' on '%s' does not return a valid DBField instance.",
+                        $fieldName,
+                        get_class($this)
+                    ));
+                }
```

Comment:
Super minor semantics request:

Can we flip the condition and throw the exception in the `if` statement? Then we can drop the `else`.